### PR TITLE
[ADD] contract_digitized_signature

### PR DESCRIPTION
**Pull request by cubells**
[WIP] Depends on
- [x] https://github.com/OCA/web/pull/666  
- [x] https://github.com/OCA/contract/pull/81

New module that allows to add digitized signature to contracts and send by mail.

cc @Tecnativa

### CODE REVIEW

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Mail template and button should be included in main `contract` module.

File: contract_digitized_signature/README.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+This module depends on :
+
+* web_widget_digitized_signature
+
+This module is part of the OCA/web suite.
+
+Configuration
+=============
+
+#. Go to *Settings > Users > Users* and set a digital signature for a user.
```
Comment (pedrobaeza) [inline contract_digitized_signature/README.rst]: These instructions are not up to date with current features.

File: contract_digitized_signature/report/report_contractorder.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" ?>
```
Comment (pedrobaeza) [inline contract_digitized_signature/report/report_contractorder.xml]: Split this out and put it on main contract module.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: contract_digitized_signature/report/report_contractorder.xml (hunk 2 above)
Comment (cubells) [inline contract_digitized_signature/report/report_contractorder.xml]: done.

File: contract_digitized_signature/README.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+This module depends on :
+
+* web_widget_digitized_signature
+
+This module is part of the OCA/web suite.
+
+Configuration
+=============
+
+#. Go to *Sales > Sales > Contracts* and print a contract order. Signature
```
Comment (pedrobaeza) [inline contract_digitized_signature/README.rst]: Update these instructions with the missing parts (where to sign)

File: contract_digitized_signature/models/account_analytic_account.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _name = 'account.analytic.account'
+    _inherit = ['account.analytic.account', 'mail.thread']
+
+    user_id = fields.Many2one(
```
Comment (pedrobaeza) [inline contract_digitized_signature/models/account_analytic_account.py]: Why adding this field?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: contract_digitized_signature/README.rst (hunk 3 above)
Comment (cubells) [reply contract_digitized_signature/README.rst]: Done.

**Comment by rafaelbn** [comment]
Please @cubells could you rebase? All dependencies are merged. Thanks

**Review by tafaRU — APPROVED** [review]
Once https://github.com/OCA/contract/pull/79#issuecomment-326318001 will be addressed. Thanks!

**Comment by pedrobaeza** [comment]
@cubells please finish this

**Comment by rafaelbn** [comment]
Hi @cubells , runbot fails. Please review.

I've tried to install the module an I get this error:

```


Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/openerp/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/openerp/http.py", line 687, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/openerp/http.py", line 323, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/openerp/http.py", line 316, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/openerp/http.py", line 966, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/openerp/http.py", line 516, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/odoo-9.0/addons/web/controllers/main.py", line 899, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/odoo-9.0/addons/web/controllers/main.py", line 887, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/.repo_requirements/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/.repo_requirements/odoo/openerp/addons/base/module/module.py", line 459, in button_immediate_install
    return self._button_immediate_function(cr, uid, ids, self.button_install, context=context)
  File "/.repo_requirements/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/.repo_requirements/odoo/openerp/addons/base/module/module.py", line 534, in _button_immediate_function
    registry = openerp.modules.registry.RegistryManager.new(cr.dbname, update_module=True)
  File "/.repo_requirements/odoo/openerp/modules/registry.py", line 386, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/.repo_requirements/odoo/openerp/modules/loading.py", line 340, in load_modules
    loaded_modules, update_module)
  File "/.repo_requirements/odoo/openerp/modules/loading.py", line 239, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/.repo_requirements/odoo/openerp/modules/loading.py", line 156, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/.repo_requirements/odoo/openerp/modules/loading.py", line 98, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 849, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 932, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 802, in parse
    self._tags[rec.tag](self.cr, rec, de, mode=mode)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 775, in _tag_template
    return self._tag_record(cr, record, data_node)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 697, in _tag_record
    f_val = self.id_get(cr, f_ref)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 780, in id_get
    res = self.model_id_get(cr, id_str, raise_if_not_found)
  File "/.repo_requirements/odoo/openerp/tools/convert.py", line 791, in model_id_get
    raise_if_not_found=raise_if_not_found)
  File "/.repo_requirements/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/.repo_requirements/odoo/openerp/addons/base/ir/ir_model.py", line 1002, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(cr, uid, xmlid)[1:3]
  File "/.repo_requirements/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in xmlid_lookup
  File "/.repo_requirements/odoo/openerp/tools/cache.py", line 85, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/.repo_requirements/odoo/openerp/addons/base/ir/ir_model.py", line 992, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % (xmlid))
ParseError: "External ID not found in the system: contract.report_contract_document" while parsing None:4, near
<data inherit_id="contract.report_contract_document">
        <xpath expr="//div[@id='invoice_info']" position="after">
            <div class="row">
                <div id="signature">
                    <p>
                        <strong>Customer acceptance:</strong>
                        <span t-if="o.customer_signature">
                            <img class="image" t-att-src="'data:image/png;base64,%s' % o.customer_signature" style="border:auto;"/>
                        </span>
                    </p>
                </div>
            </div>
        </xpath>
    </data>
```

**Comment by cubells** [comment]
Runbot is not taking into account pr #81 

I'll close this and reopen another one to pass all tests.

**Comment by cubells** [comment]
Related: #102

**Pull request closed without merging**